de.metas.vertical.pharma/src/main/java/de/metas/impexp/product/PharmaProductImportProcess.java

```diff
@@ -94,26 +94,44 @@ protected void updateAndValidateImportRecords()
 	protected ImportRecordResult importRecord(final IMutable<Object> state, final I_I_Pharma_Product importRecord) throws Exception
 	{
 		final org.compiere.model.I_M_Product existentProduct = productDAO.retrieveProductByValue(getCtx(), importRecord.getA00PZN());
-		final I_M_Product product;
-		final boolean newProduct;
-		if (existentProduct == null || importRecord.getM_Product_ID() <= 0)
+
+		final String operationCode = importRecord.getA00SSATZ();
+		if (DEACTIVATE_OPERATION_CODE.equals(operationCode) && existentProduct != null)
 		{
-			newProduct = true;
-			product = createProduct(importRecord);
+			deactivateProduct(existentProduct);
+			return ImportRecordResult.Updated;
 		}
-		else
+		else if (!DEACTIVATE_OPERATION_CODE.equals(operationCode))
 		{
-			newProduct = false;
-			product = updateProduct(importRecord, existentProduct);
-		}
+			final I_M_Product product;
+			final boolean newProduct;
+			if (existentProduct == null || (existentProduct == null && importRecord.getM_Product_ID() <= 0))
```

Comment:
it's not clear to me what this expression is about..
if i get it right, it can be simplified to 
`(existentProduct == null || importRecord.getM_Product_ID() <= 0)`
or is there a `!` missing somewhere?

also, why not assign the result of this boolean expression directly to `final boolean newProduct;`?

Reply:
@metas-ts no, is not the same;
I had ```(existentProduct == null || importRecord.getM_Product_ID() <= 0)```
And now I modified to ```(existentProduct == null || (existentProduct == null && importRecord.getM_Product_ID() <= 0))```.
The first condition does not verify correctly this case:
* The product exists in the db , but wasn't match yet in I_Pharma_Product
Bur thinking now, actually this should not be a problem. Because should have been already solved by this ```org.adempiere.impexp.product.MProductImportTableSqlUpdater.dbUpdateProductsByValue(String)```
So I'll check why that is not doing what suppose to do!